### DISCUSSION TITLE
Enhancement for TeamTalkPy

### DISCUSSION THREAD

**Discussion by lbk2907** [BearWare/TeamTalk5#1781] [Ideas]
### Description

Greetings to the developers of TeamTalkPy! Your library provides valuable functionality for Python developers looking to create chat applications and scripts. However, as a developer who has utilized your library, I have faced a significant issue with the limited functionality that restricts user and channel management capabilities. I understand that this is a complex task and I appreciate the work that has already been done. Nevertheless, I wanted to share my experience and the challenges that I have faced with the hope that it could help improve the library in the future.
As a developer utilizing TeamTalkPy library, I've faced a significant issue with the limited functionality that restricts user management capabilities. Although it offers some useful functions, such as joining a channel, sending messages, connecting to a server, etc, it lacks critical functions such as creating, deleting, or moving users, making it impossible to manage user accounts or kick or move users.
The absence of such user management functions poses a major obstacle for developers who want to develop custom applications or scripts that interact with TeamTalkPy. For instance, developing a chatbot that can manage user accounts or enforce certain rules requires access to functions for creating and deleting users. Without these functions, such bots' capabilities will be limited.
Moreover, the lack of user management functions makes it challenging to develop custom client applications that enable users to manage their accounts or move to different channels. Without these functions, client applications would also be limited in their functionality.
Overall, the missing user management functions represent a significant limitation for developers utilizing TeamTalkPy. Despite the library's useful functions, the absence of these critical functions makes it difficult to develop custom applications or scripts that require advanced user management capabilities.
Apart from the missing user management functions, there are other functions that are also absent from the current list of functions, such as managing channels, creating or deleting channels, retrieving information about channels, or changing channel properties, etc. Without these functions, it's difficult to build custom applications or scripts that require advanced management capabilities.
To further illustrate the issue, consider the example of the _DoLoginEx function that allows users to log in to a server. Python provides a corresponding higher-level function called doLogin, which accepts login information and calls _DoLoginEx to perform the actual login. Similarly, other functions such as _GetVersion, _SetLicenseInformation, _DoJoinChannelByID, _DoChangeNickname, and _DoTextMessage have corresponding Python functions in the library.
However, for functions like _DoKickUser, _DoMoveUser, _DoListUserAccounts, _DoNewUserAccount, and _DoDeleteUserAccount, etc, there are currently no higher-level Python functions provided by TeamTalkPy. This means that developers are compelled to write their Python functions to call these lower-level functions, which can be a time-consuming and error-prone process.
This issue can be particularly challenging for developers who don't have experience with C++ and may struggle to write the necessary Python functions to call the lower-level functions. This could potentially limit the pool of developers who are able to utilize the library to those with C++ expertise, excluding developers who primarily work with other programming languages. While there are resources available for learning C++ and writing Python functions that call C++ functions, it still presents an additional barrier for some developers.
Despite these challenges, I remain grateful for the work you have done in creating TeamTalkPy. Your library has enabled me to build powerful applications that leverage its unique capabilities, and I look forward to continued advancements and improvements in the future. Thanks.

@yukie-nobuharu

### Application

- [ ] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkServer

**Comment by muamel1** [comment]
You are right chat gpt

**Comment by bear101** [comment]
There's been a couple of more commits on TeamTalk5.py

**Comment by Nardol** [comment]
I opened #1812 to add some moderation related functions to TeamTalkPy.
And yesterday `doKickUser` and `doRemoveChannel` were merged.

If some function are missing, I can take some time to add them so as this discussion is here, suggestions are welcome for me.
